File: acm.timus.ru/Pascal/problem1585.pas
program problem1585;
uses Math;
const
        names:array[1..3] of string[16] = ('Emperor Penguin', 'Macaroni Penguin', 'Little Penguin');
        counts:array[1..3] of word = (0,0,0);
var
        i,n:integer;
        str:string[16];
begin
        readln(n);

        for i:=1 to n do begin
                readln(str);
                case str[1] of
                        'E':counts[1]:=counts[1]+1;
                        'M':counts[2]:=counts[2]+1;
                else
                        counts[3]:=counts[3]+1
                end;
        end;
        if (counts[1]>counts[2]) and (counts[1]>counts[3]) then
                write(names[1])
        else if (counts[2]>counts[1]) and (counts[2]>counts[3]) then
                write(names[2])
        else
                write(names[3])
end.

File: acm.timus.ru/Java/problem1585.java

import java.io.InputStreamReader;
import java.io.BufferedReader;
import java.io.StreamTokenizer;
import java.io.IOException;

public class problem1585 {

    public static void main(String[] args) throws IOException {
        BufferedReader in = new BufferedReader(new BufferedReader(new InputStreamReader(System.in)));
        StreamTokenizer st = new StreamTokenizer(in);

        st.nextToken();
        int N = (int) st.nval;
        int[] a = new int[3];
        in.readLine(); //bez etogo WA#3

        for (int i = 0; i < N; i++) {
            switch (in.read()) {
                case (int) 'E':
                    a[0]++;
                    break;
                case (int) 'L':
                    a[1]++;
                    break;
                case (int) 'M':
                    a[2]++;
                    break;
            }
            in.readLine();
        }
        int max = Math.max(Math.max(a[0], a[1]), a[2]);
        if (max == a[0]) {
            System.out.print("Emperor Penguin");
        } else if (max == a[1]) {
            System.out.print("Little Penguin");
        } else {
            System.out.print("Macaroni Penguin");
        }
    }
}
